src/gpio/gpio_chardev.c

```diff
@@ -376,56 +379,85 @@ mraa_get_line_values(int line_handle, unsigned int num_lines, unsigned char outp
 
 
 mraa_boolean_t
-mraa_is_gpio_line_kernel_owned(mraa_gpiod_line_info *linfo)
+mraa_is_gpio_line_kernel_owned(mraa_gpiod_line_info* linfo)
 {
     return (linfo->flags & GPIOLINE_FLAG_KERNEL);
 }
 
 mraa_boolean_t
-mraa_is_gpio_line_dir_out(mraa_gpiod_line_info *linfo)
+mraa_is_gpio_line_dir_out(mraa_gpiod_line_info* linfo)
 {
     return (linfo->flags & GPIOLINE_FLAG_IS_OUT);
 }
 
 mraa_boolean_t
-mraa_is_gpio_line_active_low(mraa_gpiod_line_info *linfo)
+mraa_is_gpio_line_active_low(mraa_gpiod_line_info* linfo)
 {
     return (linfo->flags & GPIOLINE_FLAG_ACTIVE_LOW);
 }
 
 mraa_boolean_t
-mraa_is_gpio_line_open_drain(mraa_gpiod_line_info *linfo)
+mraa_is_gpio_line_open_drain(mraa_gpiod_line_info* linfo)
 {
     return (linfo->flags & GPIOLINE_FLAG_OPEN_DRAIN);
 }
 
 mraa_boolean_t
-mraa_is_gpio_line_open_source(mraa_gpiod_line_info *linfo)
+mraa_is_gpio_line_open_source(mraa_gpiod_line_info* linfo)
 {
     return (linfo->flags & GPIOLINE_FLAG_OPEN_SOURCE);
 }
 
+static int
+dir_filter(const struct dirent* dir)
+{
+    return !strncmp(dir->d_name, CHIP_DEV_PREFIX, strlen(CHIP_DEV_PREFIX));
+}
+
 int
 mraa_get_number_of_gpio_chips()
 {
-    int num_chips = 0;
-    DIR* dev_dir;
-    struct dirent* dir;
-    const unsigned int len = strlen(CHIP_DEV_PREFIX);
-
-    dev_dir = opendir(DEV_DIR);
-    if (dev_dir) {
-        while ((dir = readdir(dev_dir)) != NULL) {
-            if (!strncmp(dir->d_name, CHIP_DEV_PREFIX, len)) {
-                num_chips++;
-            }
-        }
-        closedir(dev_dir);
-    } else {
-        syslog(LOG_ERR, "[GPIOD_INTERFACE]: opendir() error");
+    int num_chips;
+    struct dirent** dirs;
+
+    num_chips = scandir("/dev", &dirs, dir_filter, alphasort);
```

Comment:
Requires glibc > 2.15 but I don't feel like a version check is explicitly required.